### ISSUE TITLE
Console output preview

### ISSUE THREAD

**Issue by webmandesign** [absalomedia/vagrant-manager#10]
**Is your feature request related to a problem? Please describe.**
First, thank you for this Vagrant Manager! I try to use it as a replacement for http://vagrantmanager.com/windows/ I don't want to compare or anything, but I miss one useful feature in your product, which is **displaying of console output**.

**Describe the solution you'd like**
As can be seen in animated screenshot at http://vagrantmanager.com/windows/ every time you run a Vagrant task, such as `vagrant up` and/or `vagrant halt` (and others), there opens a "console" popup window with an output of the task/action.

I would very much appreciate including similar feature in your software as I personally rely especially on `vagrant halt` output for confirmation of database backup. I also think it could help solving any issue that may arise during the unsuccessful task run.

Could similar popup with console output be added to your software, please?
Or maybe just some log file management system doing similar functionality except directly writing the console output into a `log.txt` file, for example?

Thank you!

**Comment by absalomedia** [comment]
Creating a console output might be possible but that depends on node-vagrant,  which I use to manage each Vagrant instance,  as I’m unsure what output it’s returning back.

I wouldn’t have such a console as default due to the way Electron manages app screens across dIfferent platforms

I’ve already got basic logging running through Vagrrant Manager. Where it logs to depends on the platform - OSX, Windows or Linux

**Comment by webmandesign** [comment]
Hi @absalomedia,

Thank you for your answer.

I expected this Vagrant Manager to log some data somewhere and I actually found a `%appdata%/Vagrant Manager` folder on my Windows 10 machine. There is a `logs` subfolder there too, but this is empty.

And I'm actually not sure whether this `%appdata%/Vagrant Manager` folder is related to your app or the other one I used before - they have the same names ;)

If this Vagrant Manager produces some logs, could you please let me know the location for Windows OS? Is there some documentation where this is mentioned maybe?

Could there be an option in app settings to specify where the log file should be created please?
And maybe even a "link" in the app right click menu to access the log file?

Thanks very much!

**Comment by absalomedia** [comment]
The current logging system uses Electron-log.

By default it writes logs to the following locations:

on Linux: ~/.config/Vagrant Manager/log.log
on macOS: ~/Library/Logs/Vagrant Manager:log.log
on Windows: %USERPROFILE%\AppData\Roaming\Vagrant Manager\log.log

I’m expanding how much interaction is captured by the log & also looking at a possible solution for your console output.

Most app development paths tend to get more & more options over tIme,  harming the overall size, speed & experience of those apps. I’m aiming for efficient bruta minimalist design instead

**Comment by webmandesign** [comment]
Hi,

Thanks for the quick reply.

The `%USERPROFILE%\AppData\Roaming\Vagrant Manager` folder is actually exactly the same as the `%appdata%/Vagrant Manager`. However, I can't see any `log.log` file in there. Here is a structure of that folder on my computer:
```
[dir] GPUCache
⊢ [file] data_0
⊢ [file] data_1
⊢ [file] data_2
⊢ [file] data_3
⊢ [file] index
[dir] logs
[file] lockfile
[file] Preferences
```
Is this coming from this Vagrant Manager app, do you recognize the folder structure? Maybe it belongs to previous app I was using. The `logs` subfolder is empty and as you can see, there is no `log.log` file anywhere there. I will try to reinstall this Vagrant Manager app (and remove the folder) later today to see if that helps and will let you know of the outcome.

I completely understand your decision for very minimal design of this app and I actually welcome that. However, for dealing with Vagrant boxes I think having an output of the Vagrant task process is vital to track and resolve any potential issues that may occur with a particular Vagrant box.

**Comment by webmandesign** [comment]
OK, I've uninstalled Vagrant Manager, deleted the `%appdata%/Vagrant Manager` folder and installed Vagrant Manager back again. It created a new `%appdata%/Vagrant Manager` folder with this structure:
```
[dir] logs
[file] lockfile
```

Again, `logs` folder is empty. There is no `log.log` file anywhere.

So, could the log file functionality be added to the app please? Thank you!

**Comment by absalomedia** [comment]
With version 2.2, the log output folder has been changed to VagrantManager (one word) to stop clashing with the Lanayo Technologies edition of it. It will still output a file named log.log. It will also be much more detailed.

There's a new setting - enabling console view - that allows Vagrant to output the responses from the various vagrant commands & actions to a modal box linked into the UI of each platform (Windows, OSX, Linux). The only trade off with this console view is that once the modal is shown, it pauses the application process until that modal is closed.

**Comment by webmandesign** [comment]
Great! Thank you very much!

The paused application process is perfectly fine with me as I always check the output before starting/finishing work with Vagrant.

If you need some beta testing of the functionality, I'm in :)

Thank you again!

**Comment by webmandesign** [comment]
Oh, sorry, I can see the version 2.2 was released now, so I'll go ahead and upgrade and test ;)

**Comment by webmandesign** [comment]
I've just tested the new version 2.2 and it works fine, thank you.

Just some suggestions for improvement:

1. Currently the console window seems to open only after the Vagrant task is ended. Would be great to keep the window open from the start and to see it doing something (logging the process). Now I was confused and I thought it actually doesn't work until the window appeared after the Vagrant task ended. If that's possible to implement, it would be great, if not, I'm fine with this too :)
2. Could there be some "console styling" applied on the window output please? Especially some monospace font. Thank you!

**Comment by absalomedia** [comment]
@webmandesign I'll work through these:

1. The console output is dependent on [node-vagrant](https://github.com/edin-m/node-vagrant) & I have no way of providing a stream instead of a response to whatever node-vagrant supplies me with. One of the reasons I moved the Vagrant management from [Vaibhav](https://github.com/vaibhav-sidapara/vagrant-manager) to the Node edition, as part of my fork, was to simplify the overhead required. It also resolved most of the memory leak issues I was finding in their original code. 
2. Vagrant Manager being based on Electron means whilst there may be ways to set [default fonts](https://github.com/electron/electron/issues/4132)  , there are dependency issues based on Chromium & the fact I can't set a Monospace font just for one window.

Both of these are dependency issues with other packages, be it node-vagrant or Electron itself.

**Comment by webmandesign** [comment]
Thank you for explanation @absalomedia

**Comment by webmandesign** [comment]
Hi,

I've noticed an issue related to the new settings option. The "Enable console" option resets itself to unchecked state every time I run the app. Could this be saved permanently so I don't have to enable it all the time please?

Or should I create a new issue for that?

Thanks!

**Comment by absalomedia** [comment]
Best to create a new issue with it. I'll also try to replicate the issue sometime today as well.
